catalog_tools/utils/binning.py

```diff
@@ -68,6 +68,5 @@ def bin_magnitudes(mags: list, delta_x: float = 0.1) -> list:
         List of values rounded to the given precision.
     """
 
-    mags_array = np.array(mags)
-    rounded_array = bin_to_precision(mags_array, delta_x)
-    return list(rounded_array)
+    rounded_array = bin_to_precision(mags, delta_m)
+    return rounded_array
```

Comment:
With the changes like this, it might not make sense to have bin_to_precision and bin_magnitudes (they do exactly the same..) should we only keep one?